### PR TITLE
Generalize Beads prefix configuration and remove hard-coded at assumptions

### PR DESCRIPTION
**Pull request by shaug**
# Summary

- Make Beads issue-prefix selection a project-scoped, configurable part of setup instead of relying on a hard-coded `at` prefix.
- Add deterministic prefix suggestion and collision handling so local multi-project Atelier usage avoids namespace conflicts.
- Propagate configured prefixes through runtime flows (planner/worker/startup/lifecycle/export) and Dolt naming while preserving compatibility for existing stores.

# Changes

- Added `beads.prefix` support in config/models, including normalization, validation, suggestion (`tuber-service -> ts`), and collision-aware fallback (`ts2`, `ts3`, ...).
- Added `--beads-prefix` support to `atelier init` and `atelier new`, and threaded the value through project initialization.
- Replaced hard-coded `at:*` assumptions with prefix-aware label/prefix helpers across planner, worker, startup, GC, status, and export flows.
- Updated lifecycle label checks to be namespaced-label aware rather than prefix-literal checks.
- Derived default Dolt DB naming from configured prefix + project identity, with compatibility guards for existing database directories.
- Updated AGENTS templates and docs for prefix-aware behavior and added a migration guide for existing projects.
- Added/updated tests covering prefix derivation, collision handling, CLI wiring, runtime consumption, and migration-related behavior.

# Testing

- `just format`
- `just lint`
- `just test`

## Tickets
- None

# Risks / Rollout

- Existing projects defaulting to `at` continue to work via compatibility paths; migration steps are documented for explicit prefix moves.
- Runtime label handling now depends on configured prefix resolution and suffix-aware matching, so rollout risk is concentrated in lifecycle/GC edge-cases already covered by updated tests.

# Notes

- This is intentionally a broad but single-scope changeset to remove hard-coded prefix assumptions end-to-end and make prefix configuration first-class.

### CODE REVIEW

File: src/atelier/worker/session/runner.py [hunk 1]
```diff
@@ -90,7 +90,14 @@ def _find_active_root_branch_conflicts(
     repo_root: Path,
 ) -> list[dict[str, object]]:
     issues = beads.run_bd_json(
-        ["list", "--label", "at:epic", "--all", "--limit", "0"],
+        [
+            "list",
+            "--label",
+            beads_runtime.issue_label("epic", beads_root=beads_root),
```
Comment (shaug) [inline src/atelier/worker/session/runner.py]: This query only searches the active prefix label. The migration doc says runtime lookup stays compatibility-safe for legacy `at:*` labels during migration, but this path (and `_global_changeset_records` in `worker/review.py`) will miss legacy-labeled epics if prefix repair is partial. Can we use `issue_label_candidates("epic", ...)` + de-duplication here, similar to `beads.list_epics`?
